protokube/pkg/protokube/kube_boot.go

```diff
@@ -101,8 +102,23 @@ func (k *KubeBoot) Init(volumesProvider Volumes) {
 
 // RunSyncLoop is responsible for provision the cluster
 func (k *KubeBoot) RunSyncLoop() {
+	ctx := context.Background()
+
+	for {
+		client, err := k.Kubernetes.KubernetesClient()
+		if err != nil {
+			klog.Warningf("could not create kubernetes client: %v", err)
+		}
+
+		_, err = client.CoreV1().Namespaces().Get(ctx, "kube-system", metav1.GetOptions{})
+		if err == nil {
+			klog.Info("connected to the apiserver")
+			break
+		}
+		klog.Infof("Could not connect to apiserver: %v", err)
```

Comment:
Nit: something to indicate that this is normal(ish) might be helpful here.  e.g. "polling for apiserver readiness; not yet ready: %v" or something like that?